src/tally.F90

```diff
@@ -395,6 +402,195 @@ subroutine score_general(p, t, start_index, filter_index, i_nuclide, &
+              if (dg_filter > 0) then
+
+                ! Loop over all delayed group bins and tally to them
+                ! individually
+                do d_bin = 1, t % filters(dg_filter) % n_bins
+
+                  ! Get the delayed group for this bin
+                  d = t % filters(dg_filter) % int_bins(d_bin)
+
+                  ! Compute the yield for this delayed group
+                  yield = yield_delayed(nuc, p % E, d)
+
+                  ! Compute the score and tally to bin
+                  score = p % absorb_wgt * yield * micro_xs(p % event_nuclide) &
+                       % fission * nu_delayed(nuc, p % E) / &
+                       micro_xs(p % event_nuclide) % absorption
+                  call score_fission_delayed_dg(t, d_bin, score, score_index)
+                end do
+                cycle SCORE_LOOP
+              else
+                ! If the delayed group filter is not present, compute the score
+                ! by multiplying the absorbed weight by the fraction of the
+                ! delayed-nu-fission xs to the absorption xs
+                score = p % absorb_wgt * micro_xs(p % event_nuclide) &
+                     % fission * nu_delayed(nuc, p % E) / &
+                     micro_xs(p % event_nuclide) % absorption
+              end if
+            end if
+          else
+            ! Skip any non-fission events
+            if (.not. p % fission) cycle SCORE_LOOP
+            ! If there is no outgoing energy filter, than we only need to
+            ! score to one bin. For the score to be 'analog', we need to
+            ! score the number of particles that were banked in the fission
+            ! bank. Since this was weighted by 1/keff, we multiply by keff
+            ! to get the proper score. Loop over the neutrons produced from
+            ! fission and check which ones are delayed. If a delayed neutron is
+            ! encountered, add its contribution to the fission bank to the
+            ! score.
+
+            ! Check if the delayed group filter is present
+            if (dg_filter > 0) then
+
+              ! Loop over all delayed group bins and tally to them individually
+              do d_bin = 1, t % filters(dg_filter) % n_bins
+
+                ! Get the delayed group for this bin
+                d = t % filters(dg_filter) % int_bins(d_bin)
+
+                ! Compute the score and tally to bin
+                score = keff * p % wgt_bank / p % n_bank * p % n_delayed_bank(d)
+                call score_fission_delayed_dg(t, d_bin, score, score_index)
+              end do
+              cycle SCORE_LOOP
+            else
+
+              score = ZERO
+
+              ! Loop over all delayed groups and accumulate the contribution
+              ! from each group
+              do d = 1, nuclides(p % event_nuclide) % n_precursor
+                score = score + keff * p % wgt_bank / p % n_bank * &
+                     p % n_delayed_bank(d)
```

Comment:
The factors `keff * p % wgt_bank / p % n_bank` are constant and thus don't really need to be in the loop. A simpler way might be:
```fortran
score = keff * p % wgt_bank / p % n_bank * &
     sum(p % n_delayed_bank)
```